Document files promise content attribute

lluiscampos opened the pull request.

nickanderson commented on reference/promise-types/files.markdown:
````diff
@@ -1404,6 +1404,30 @@ comparison or linking operations).
+
+**Example:**
+
+```cf3
````
Instead of embedding a snippet example like this, I thinkit's better to dynamically include the example from core.

https://github.com/cfengine/documentation/blob/master/README.md#quoting-policy-files

e.g. https://github.com/cfengine/documentation/blob/master/reference/promise-types/files.markdown#missing_ok

lluiscampos replied:
Thanks for this tip! I wasn't aware of it. I will rework.

lluiscampos replied:
Fixed :+1:

nickanderson commented on reference/promise-types/files.markdown:
```diff
@@ -1404,6 +1404,30 @@ comparison or linking operations).
+
+**History:** Was introduced in 3.16.0
+
+**Note:** You cannot `content` in combination with the other edit operations
```
What does the agent do if you do combine those attributes that are mutually exclusive?

lluiscampos replied:
@nickanderson That is a good question. I am writing some acceptance tests for this. My idea was that `cf-agent` should log the error message and exit with error. Currently it will print something like:
```
error: A file promise with content attribute cannot have edit operations
```

lluiscampos replied:
See acceptance tests at https://github.com/cfengine/core/pull/4219

nickanderson commented on reference/promise-types/files.markdown:
```diff
@@ -1404,6 +1404,25 @@ comparison or linking operations).
 
+### content
+
+**Description:** Content to write into file
```
I think this should rather say `Complete content the file should contain`

lluiscampos commented:
> I just changed it directly myself.

@nickanderson Thanks! I squashed it now for nicer Git history. I am still waiting for the review in the implementation PR, anyway.
